src/persist-client/src/fetch.rs

```diff
@@ -627,12 +636,38 @@ where
                 continue;
             }
 
-            let k = self.metrics.codecs.key.decode(|| K::decode(k));
-            let v = self.metrics.codecs.val.decode(|| V::decode(v));
+            let k = self.metrics.codecs.key.decode(|| match key.take() {
+                Some(mut key) => match K::decode_from(&mut key, k, &mut self.key_storage) {
+                    Ok(()) => Ok(key),
+                    Err(err) => Err(err),
+                },
+                None => K::decode(k),
+            });
+            let v = self.metrics.codecs.val.decode(|| match val.take() {
+                Some(mut val) => match V::decode_from(&mut val, v, &mut self.val_storage) {
+                    Ok(()) => Ok(val),
+                    Err(err) => Err(err),
+                },
```

Comment:
Uncovered

Reply:
Expected since in production `val` is the `()` type, and we don't pass a buffer to be reused.